scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,305 @@
+                      "input_auth_domain_name": "NA",
+                      "auth_domain_email": "Incomplete",
+                      "auth_domain_creation_error": "NA",
+                      "email_added_to_AD": "Incomplete",
+                      "add_email_to_AD_error": "NA",
+                      "workspace_link": "Incomplete",
+                      "workspace_creation_error": "NA",
+                      "workspace_ACLs": "Incomplete",
+                      "workspace_ACLs_error": "NA",
+                      "workspace_setup_status": "Failed"}
+
+    # start authorization domain
+    auth_domain_name = workspace['auth_domain_name']
+    workspace_dict["input_auth_domain_name"] = auth_domain_name
+    auth_domain_status = create_auth_domain(auth_domain_name)
+
+    if auth_domain_status != 201:  # AD creation fail - abort setup
+        workspace_dict["auth_domain_creation_error"] = auth_domain_status  # update error in dict
+        return workspace_dict
+
+    # AD creation success - continue setup
+    workspace_dict["auth_domain_email"] = f"{auth_domain_name}@firecloud.org"  # update dict with AD email
+    # add members to AD
+    anvil_admin_group_name = "anvil-admins@firecloud.org"
+    add_member_status = add_member_to_auth_domain(auth_domain_name, anvil_admin_group_name, "ADMIN")
+
+    if add_member_status != 204:  # adding member to AD fail - abort setup
+        workspace_dict["add_email_to_AD_error"] = auth_domain_status  # update error in dict
+        return workspace_dict
+
+    # adding member to AD success - continue setup
+    workspace_dict["email_added_to_AD"] = anvil_admin_group_name  # update dict with member added to AD
+
+    # workspace creation if AD set up succeeds
+    workspace_name = workspace["workspace_name"]
+    workspace_dict["input_workspace_name"] = workspace_name
+    create_ws_request = make_create_workspace_request(workspace_name, auth_domain_name, project)  # json for API request
+
+    create_workspace_status = create_workspace(create_ws_request, workspace_name, project)  # create workspace
+
+    if create_workspace_status in [201, 200]:  # new ws created or user selected 'continue with existing ws'
+        workspace_dict["workspace_link"] = (f"https://app.terra.bio/#workspaces/{project}/{workspace_name}").replace(" ", "%20")
+        # add ACLs to workspace
+        member_status_code, member_response = add_members_to_workspace(workspace_name, auth_domain_name, project)
+
+        if member_status_code != 200:  # adding ACLs to workspace fail
+            workspace_dict["workspace_ACLs_error"] = member_response
+            return workspace_dict
+
+        # adding ACLs to workspace success
+        workspace_dict["workspace_ACLs"] = member_response  # update dict with ACL emails
+        workspace_dict["workspace_setup_status"] = "Success"  # final workspace setup step
+    else:  # if "N" (user does not want to continue with existing workspace) or other error
```

Comment:
this is small - but to be consistent with your previous code, where you use the `if` statements to check for errors, you could flip this around and on line 252 say 
```
if create_workspace_status not in [201, 200]: 
    workspace_dict["workspace_creation_error"] = create_workspace_status  # update error in dict
    return workspace_dict
```
and then proceed with the rows 253 onward (again not in an else statement)